### PR TITLE
[EraVM] Refactor the three other `InstrMapping`s

### LINKED ISSUE #482: I introduced that duplication because I had issues with `OperandAddrMode` as the ColField. As you refactor our `td`s anyway, may I ask if it's hard to get rid of `OperandAM`?
I introduced that duplication because I had issues with `OperandAddrMode` as the ColField. As you refactor our `td`s anyway, may I ask if it's hard to get rid of `OperandAM`?

_Originally posted by @akiramenai in https://github.com/matter-labs/era-compiler-llvm/pull/448#discussion_r1553744353_

### PR DESCRIPTION
**Pull request by atrosinenko**
Unlike mapping between different input operand kinds, the three other `InstrMapping`s only have two options each that are flipped by the mapper function.

Remove the synthetic Value field from `DestAddrModeValue` (renamed to `DestOperandMode`) class. Keep Value fields in `mod_set_flags` and `mod_swap` as these fields are meaningful for opcode computation.

Remove the `ToReg` vs. `ToRegReg` and `ToStack` vs. `ToStackReg` separation as it is not really used.

### CODE REVIEW

**Review by akiramenai — APPROVED** [review]
LGTM,
But I'm not sure if we should apply TableGen tag, as it only changes td files inside EraVM. However, I don't know what is the common practice on the matter, so I leave the decision to you.

**Comment by asl** [comment]
Looks like benchmarks changed... @sayon can we get results dump?

**Comment by atrosinenko** [comment]
@asl Specifically, looks like the results **were updated** by the bot - probably after I merged #601 without waiting for the CI once again after changing commit message only.

**Comment by asl** [comment]
@atrosinenko Ah, oh. Can you rebase then?

**Comment by atrosinenko** [comment]
> Ah, oh. Can you rebase then?

@asl Strictly speaking, it is already rebased onto current HEAD of `main` branch :) Moreover, the benchmark results in #601 were updated the same way (and have seemingly the same delta values). Maybe restarting some CI job would help, but I'm not sure what to restart.

**Comment by asl** [comment]
@antonbaliasnikov @akiramenai @sayon Can you help?

**Comment by atrosinenko** [comment]
Finally, after pushing more commits to `main` and rebasing this PR, the benchmark results were recomputed again and now they are as expected.

**Pull request merged**
